src/Xrd/XrdPollPoll.icc

```diff
@@ -274,7 +276,15 @@ void XrdPollPoll::Start(XrdSysSemaphore *syncsem, int &retcode)
 
 // Now do the main poll loop
 //
-   do {do {numpolled = poll(PollTab, PollTNum, -1);}
+   std::vector<struct pollfd> PollTabCopy;
+   do {// Duplicate the polling table so we don't need to hold the PollMutex
+       // while we are sleeping in the poll()
+       PollMutex.Lock();
+       PollTabCopy.resize(PollTNum);
+       memcpy(PollTabCopy.data(), PollTab, sizeof(struct pollfd) * PollTNum);
```

Comment:
This is going to be really slow. Let's add a commit to fix the poller on macOS, but I suggest having two `PollTab` structs allocated, then swapping them at each call to `poll(PollTab, PollTNum, -1)` instead of copying over each time and calling `PollTabCopy.resize(PollTNum);` inside a loop. Please keep only this change and the addition of the test in this pull request and split unrelated changes (modernize linking of libcurl) into a separate pull request.

Reply:
I'm not following.

While the poll is running, the other code is making arbitrary changes to the data structure (that's part of the core of what makes the implementation thread-unsafe).  We can't swap here because the copy of the data structure fed to the `poll` function is missing an unknown number of changes.  The only option is to create a copy.

With respect to the resize -- this is typically separate from a reallocation and the "extra cost" is zeroing out any growth in the file descriptor count.

Avoiding the copy is going to require a much more significant surgery to the poller.

Reply:
Ah, scratch that -- I think I see what you mean a bit lower in the code.  Maybe a swap will work after all; I'll tinker.

Reply:
Do you plan to move from copying to swapping between two `PollTab`s?

Reply:
Unfortunately, my tinkering failed -- I wasn't able to find a way to prevent multiple threads from writing into the table while the poll is ongoing.  That means, for now, the copy is required.

I don't think a swap will work here: you need to synchronize the changes so you need to copy the data back and forth.  Luckily, the resize shouldn't mean a reallocate: the only "cost" should be the memcpy.

Reply:
Ok, since this is only for macOS in any case, let's fix now and optimize later then.

Reply:
Yes - that's roughly my thinking.  I'd love to spend more time here but, pragmatically, there are only so many hours over the weekend.

It certainly beats my prior patch we carried for CI purposes: we prevented file descriptors from being deleted from the table, meaning it slowly grew (but otherwise "behaved" correctly) throughout the course of testing.